src/R/Sql/Impl/Publish/DacPackageServices.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.ComponentModel.Composition;
+using Microsoft.R.Components.Sql.Publish;
+
+namespace Microsoft.VisualStudio.R.Sql.Publish {
+    [Export(typeof(IDacPackageServices))]
```

Comment:
But wouldn't that create hard link even if code is not called and throw on assembly loading? If code in `DacPackageServicesProvider` did `new DacPackageServices()` and that class, in turn, had code `new DacPacBuilder()` and the `DacPacBuilder` implementation called `TSqlModel` which was not present since SSDT assemblies are not installed, wouldn't that prevent package from loading?

I.e. the point is to prevent loading of `Microsoft.VisualStudio.R.Sql` when SSDT is not installed.